### ISSUE TITLE
dotnet build/publish handles solution-level dependency on Windows but not Linux

### ISSUE THREAD

**Issue by mvinca** [dotnet/cli#8749]
We have noticed that we cannot build our solution on Linux build machines, but we can on Windows build machines. After some investigation, we have created a sample project. It seems the culprit is solution-level project dependencies. Note, this is not a code project dependency. There is no compile dependency between the two units. We are using solution-level dependencies to control build order for other reasons than code dependency (mainly our "main" project is last, and it does some organization.)

In case someone reading this isn't familiar with solution-level dependencies, in Visual Studio:
1. Right click on the Solution in Solution Explorer
2. Choose "Project Dependencies..."
3. Here you can check a box to add additional build order dependencies that aren't related to code dependencies.
4. These dependencies are recorded in the .sln file, but do not affect .csproj files.

## Steps to reproduce

1. Clone https://github.com/mvinca/DotNetCliSolutionDependency onto a Windows Machine with .NET SDK 2.1.4
2. type "dotnet publish ServerWithPlugins.sln"
3. This works
4. Clone https://github.com/mvinca/DotNetCliSolutionDependency onto a LinuxMachine with .NET SDK 2.1.4
5. type "dotnet publish ServerWithPlugins.sln"
6. This fails, see "Actual Behavior" below
7. Edit ServerWithPlugins.sln and remove the solution dependency
8. type "dotnet publish ServerWithPlugins.sln"
9. This works

## Expected  behavior
After Step 2) Publish succeeds and file is copied into plug-ins folder
After Step 5) Publish succeeds and file is copied into plug-ins folder, consistent with Windows

## Actual behavior
After Step 2) Publish succeeds and file is copied into plug-ins folder
After Step 5) Publish fails, seemingly due to parse error of the .sln file

### Example
`/home/mvinca/Documents/solutiondependencyissue/ServerWithPlugins/ServerWithPlugins/ServerWithPlugins.csproj.metaproj : error MSB4025: The project file could not be loaded. Could not find file '/home/mvinca/Documents/solutiondependencyissue/ServerWithPlugins/ServerWithPlugins/ServerWithPlugins.csproj.metaproj'.`

## Environment data
`dotnet --info` output:

```
.NET Command Line Tools (2.1.4)

Product Information:
 Version:            2.1.4
 Commit SHA-1 hash:  5e8add2190

Runtime Environment:
 OS Name:     centos
 OS Version:  7
 OS Platform: Linux
 RID:         centos.7-x64
 Base Path:   /usr/share/dotnet/sdk/2.1.4/

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.5
  Build    : 17373eb129b3b05aa18ece963f8795d65ef8ea54
```

**Comment by livarcocc** [comment]
@rainersigwald any ideas here? Seems to be failing during the metaproj generation.

**Comment by rainersigwald** [comment]
It's probably https://github.com/Microsoft/msbuild/issues/2022 which will be fixed in 15.7/2.1.300.

**Comment by livarcocc** [comment]
Ok, closing in favor of the MSBuild one.

**Comment by mvinca** [comment]
Just confirming that if I replace \ with / throughout the .sln file as stated as a work around in Microsft/msbuild#2002 the solution file does indeed work properly on Linux.
